src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1316,7 +1316,8 @@ private BoundStatement ChainImplicitStructConstructor(MethodSymbol methodSymbol,
                 if (optimizations == OptimizationLevel.Debug && stateMachineTypeOpt != null)
                 {
                     Debug.Assert(method.IsAsync || method.IsIterator);
-                    GetStateMachineSlotDebugInfo(moduleBuilder.GetSynthesizedFields(stateMachineTypeOpt), out stateMachineHoistedLocalSlots, out stateMachineAwaiterSlots);
+                    GetStateMachineSlotDebugInfo(moduleBuilder, moduleBuilder.GetSynthesizedFields(stateMachineTypeOpt), diagnostics, out stateMachineHoistedLocalSlots, out stateMachineAwaiterSlots);
```

Comment:
Should this be 'diagnosticsForThisMethod'? Or can we remove 'diagnosticsForThisMethod' and use 'diagnostics' everywhere?